setup.py

```diff
@@ -22,24 +22,19 @@
 import os
 import re
 import sys
+import subprocess
 
 from setuptools import find_packages, setup
 
 PKG = "hy"
 VERSIONFILE = os.path.join(PKG, "version.py")
-verstr = "unknown"
 try:
-    verstrline = open(VERSIONFILE, "rt").read()
-except EnvironmentError:
-    pass  # Okay, there is no version file.
-else:
-    VSRE = r"^__version__ = ['\"]([^'\"]*)['\"]"
-    mo = re.search(VSRE, verstrline, re.M)
-    if mo:
-        __version__ = mo.group(1)
-    else:
-        msg = "if %s.py exists, it is required to be well-formed" % VERSIONFILE
-        raise RuntimeError(msg)
+    __version__ = subprocess.check_output([
```

Comment:
Probably it depends on exactly what method we decide to use to put a package on PyPI. However, I think this ought to include a build step, as in [this example](https://pythonhosted.org/an_example_pypi_project/setuptools.html), which uses setup.py itself to do the upload.